presto-testing-server-launcher/src/main/java/com/facebook/presto/server/testing/TestingPrestoServerLauncher.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.testing;
+
+import com.facebook.presto.server.testing.TestingPrestoServerLauncherOptions.Catalog;
+import com.facebook.presto.spi.Plugin;
+import io.airlift.command.Command;
+import io.airlift.command.Help;
+import io.airlift.command.HelpOption;
+import io.airlift.command.model.CommandMetadata;
+
+import javax.inject.Inject;
+
+import static io.airlift.command.SingleCommand.singleCommand;
+import static java.lang.Runtime.getRuntime;
+
+@Command(name = "testing_presto_server", description = "Testing Presto Server Launcher")
+public class TestingPrestoServerLauncher
+{
+    @Inject CommandMetadata commandMetadata;
+
+    @Inject
+    public HelpOption helpOption;
+
+    @Inject
+    TestingPrestoServerLauncherOptions options = new TestingPrestoServerLauncherOptions();
+
+    private static void registerServerCloseShutdownHook(final TestingPrestoServer server)
```

Comment:
Instead of registering a shutdown hook, it'd be cleaner to shut down the server in a finally block in the run method:

```java
try (TestingPrestoServer server = ...) {
    ...
    Thread.currentThread.join();
}
```